src/interactive-window/editor-integration/codeLensFactory.ts

```diff
@@ -156,7 +156,10 @@ export class CodeLensFactory implements ICodeLensFactory {
             cache.cellRanges.length &&
             this.configService.getSettings(document.uri).addGotoCodeLenses
         ) {
-            const hashes = this.getHashes();
+            const storage = this.generatedCodeStorageFactory.get({ fileUri: document.uri });
```

Comment:
Added the ability to get the hashes for a give file Uri.
Even this class can do this by going through all notebook documents, then finding a storage that has the `document.uri` in the property `all` which contians all of the Uris.

But that feels like the job of the factory to return the right storage.
Either a helper function or just move into storage, i chose to add an overload into the storage.

Reply:
Should the storage maybe perform that little .all.find check? I'd think that you pretty much would always use that line.

Reply:
can't, as we don't know which storage has the relevant yuri, hence we need to go through all storages, and only the factory had access to all via the internal map .